File: FragmentNavigation/app/src/main/java/com/example/fragmentnavigation/MenuFragment.java
package com.example.fragmentnavigation;


import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.example.fragmentnavigation.databinding.FragmentMenuBinding;
import com.example.fragmentnavigation.model.Option;

public class MenuFragment extends Fragment {

    FragmentMenuBinding binding;
    public static final int REQUEST_CODE = 1;
    Option option;
    public static String OPTION_KEY = "option" ;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentMenuBinding.inflate(inflater,container,false);
        binding.openBoxButton.setOnClickListener(v->{onOpenBox();});
        binding.optionsButton.setOnClickListener(v->{onOptions();});
        binding.aboutButton.setOnClickListener(v->{onAbout();});
        binding.exitButton.setOnClickListener(v->{onExit();});
        if (getArguments() != null){
            option = getArguments().getParcelable(OptionFragment.EXTRA_OPTION);
        }else{
            option = new Option(3,false);
        }
        return binding.getRoot();
    }

    public static MenuFragment getInstance(Bundle bundle){
        MenuFragment fragment = new MenuFragment();
        fragment.setArguments(bundle);
        return fragment;
    }



    private void onExit() {
        getActivity().finish();
    }

    private void onAbout() {
        replace(new AboutFragment());
    }

    private void onOptions() {
        Bundle bundle = new Bundle();
        bundle.putParcelable(OPTION_KEY,option);
       replace(OptionFragment.getInstance(bundle));
    }



    private void onOpenBox() {
        Bundle bundle = new Bundle();
        bundle.putParcelable(OPTION_KEY,option);
        replace(BoxFragment.getInstance(bundle));

    }

    private void replace(Fragment fragment){
        getParentFragmentManager().beginTransaction().replace(R.id.frag_container,fragment).commit();
    }
}

File: FragmentNavigationKotlin/app/src/main/java/com/example/fragmentnavigationkotlin/fragment/MenuFragment.kt
package com.example.fragmentnavigationkotlin.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.fragmentnavigationkotlin.R
import com.example.fragmentnavigationkotlin.contract.ResultListener
import com.example.fragmentnavigationkotlin.contract.navigator
import com.example.fragmentnavigationkotlin.databinding.FragmentMenuBinding
import com.example.fragmentnavigationkotlin.model.Option

class MenuFragment : Fragment() {
    private lateinit var binding:FragmentMenuBinding
    private lateinit var option:Option
    private val OPTIONS= "Option"

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentMenuBinding.inflate(inflater,container,false)
        navigator().listenerResult(Option::class.java,viewLifecycleOwner){
            this.option = it
        }
        return binding.root
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        option = if (savedInstanceState == null){
            Option.DEFAULT
        }else{
            savedInstanceState.getParcelable(OPTIONS)!!
        }
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.openBoxButton.setOnClickListener { onOpenBox() }
        binding.optionsButton.setOnClickListener { onOptions() }
        binding.aboutButton.setOnClickListener { onAbout() }
        binding.exitButton.setOnClickListener { onExit() }
    }

    private fun onAbout() {
        navigator().showAboutScreen()
    }

    private fun onOptions() {
        navigator().showOptionScreen(option)
    }

    private fun onOpenBox() {
        navigator().showBoxSelectScreen(option)
    }

    private fun onExit() {
        navigator().goBack()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putParcelable(OPTIONS,option)
    }
}